src/modalities/config/component_factory.py

```diff
@@ -0,0 +1,129 @@
+from typing import Any, Dict, List, Union
+
+from pydantic import RootModel
+
+from tests.config.hierarchical_instantiation.configs import CompConfigABC
+
+
+class ComponentFactory:
+    @staticmethod
+    def build_config(config_dict: Dict, config_types, component_names: List[str]) -> Dict[str, Any]:
+        component_dict = ComponentFactory._build_components(
+            config_dict=config_dict, config_types=config_types, component_names=component_names
+        )
+        return component_dict
```

Comment:
It seems that `build_config` only calls another method, `_build_components`. Is there a reason to have 2 separate methods?

Reply:
fixing it. Thanks!